llama_index/question_gen/llm_generators.py

```diff
@@ -56,7 +56,13 @@ def _get_prompts(self) -> PromptDictType:
     def _update_prompts(self, prompts: PromptDictType) -> None:
         """Update prompts."""
         if "question_gen_prompt" in prompts:
-            self._prompt = prompts["question_gen_prompt"]
+            output_parser = prompts["question_gen_prompt"].output_parser
+            if output_parser is None:
+                output_parser = SubQuestionOutputParser()
+            new_prompt = PromptTemplate(
```

Comment:
can just do self._prompt = PromptTemplate(***) without the need of creation of new_prompt

Reply:
changed as suggested @hatianzhang 
please check the modified code below which I pushed

```
def _update_prompts(self, prompts: PromptDictType) -> None:
    """Update prompts."""
    if "question_gen_prompt" in prompts:
        output_parser = prompts["question_gen_prompt"].output_parser
        if output_parser is None:
            output_parser = SubQuestionOutputParser()

        self._prompt = PromptTemplate(
            prompts["question_gen_prompt"].template_str,
            output_parser=output_parser
        )
```

Reply:
it LGTM!